de.dashup.model/src/main/java/de/dashup/model/builder/DashupBuilder.java

```diff
@@ -1,21 +1,29 @@
 package de.dashup.model.builder;
 
-import de.dashup.shared.Panel;
+import de.dashup.model.service.DashupService;
+import de.dashup.shared.DatabaseModels.DatabaseUser;
 import de.dashup.shared.Section;
-import de.dashup.shared.User;
+import de.dashup.shared.Widget;
+
+import java.sql.SQLException;
+import java.util.List;
 
 public class DashupBuilder {
 
-    public static String buildUsersPanels(User user) {
+    public static String buildUsersPanels(DatabaseUser user) throws SQLException {
         StringBuilder content = new StringBuilder();
-        if (user.getSections() != null) {
-            for (Section section : user.getSections()) {
+        List<Section> sections = DashupService.getInstance().loadUserLayout(user).getSections();
```

Comment:
Another reason why we have introduced the `User` class: We do not want to fetch the data every time again from the database. This is a huge lack of performance.

Reply:
Again, I did not know, that the user object and the layout is stored locally in the browser. But be aware, that a user can easily manipulate this representation this way. LoadUserLayout generates the data structure for the central dashboard as mentioned in a comment above.